File: segundos 4 lenguajes/go/JugadorNovato.go
package main

type JugadorNovato struct {
	Nombre string
	Nivel  int
}

func NewJugadorNovato() JugadorNovato {
	return JugadorNovato{"Novato", 1}
}

func NewJugadorNovatoParams(nombre string, nivel int) JugadorNovato {
	return JugadorNovato{nombre, nivel}
}

func NewJugadorNovatoNivel(nivel int) JugadorNovato {
	return JugadorNovato{"Novato", nivel}
}

func NewJugadorNovatoCopy(j JugadorNovato) JugadorNovato {
	return JugadorNovato{j.Nombre, j.Nivel}
}

File: segundos 4 lenguajes/Java/JugadorNovato.java
package Java;
class JugadorNovato {
    String nombre;
    int nivel;

    public JugadorNovato() {
        nombre = "Novato";
        nivel = 1;
    }

    public JugadorNovato(String nombre, int nivel) {
        this.nombre = nombre;
        this.nivel = nivel;
    }

    public JugadorNovato(int nivel) {
        this.nombre = "Novato";
        this.nivel = nivel;
    }

    public JugadorNovato(JugadorNovato j) {
        this.nombre = j.nombre;
        this.nivel = j.nivel;
    }
}

